pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -383,7 +383,12 @@ func streamPathToBuild(git git.Repository, in io.Reader, out io.Writer, client o
 			return nil, err
 		}
 		if stat.IsDir() {
-			info, gitErr := gitRefInfo(git, clean, "HEAD")
+			commit := "HEAD"
+			if len(options.Commit) > 0 {
+				commit = options.Commit
+			}
+			fmt.Fprintf(out, "Uploading %q at commit %q as binary input for the build ...\n", clean, commit)
```

Comment:
:+1: on removing any references to git, it'll be easier when we'll be adding other VCSs support. You man want to change the error in line 405 as well: 
`the directory %q is not a valid repository: %v`